### PR TITLE
Modify pydantic model so report endpoints accept either JSON or form data

### LINKED ISSUE #271: Report endpoints to accept POST requests with HTTP form data
This will simplify interfacing with other apps like Scout

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- Fix #271 
 
### How to test:
- Deploy on stage and use in a combo with [this scout branch](https://github.com/Clinical-Genomics/scout/pull/4553) 

### Expected outcome:
- [x] Coverage report and coverage overview reports should be created

### Review:
- [x] Code approved by HS
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner

### CODE REVIEW

**Comment by northwestwitch** [comment]
### Tested via Scout: coverage report:

image

### Coverage overview:

image

File: .github/workflows/tests_n_coverage.yml [hunk 1]
```diff
@@ -55,10 +55,12 @@ jobs:
...
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
         with:
           flags: unittests
           files: ./coverage.xml
           name: codecov-umbrella
           fail_ci_if_error: true
           verbose: true
+        env:
```
Comment (northwestwitch) [inline .github/workflows/tests_n_coverage.yml:65]: The action stopped working without

File: src/chanjo2/models/pydantic_models.py [hunk 2]
```diff
@@ -232,6 +233,42 @@ class ReportQuery(BaseModel):
     case_display_name: Optional[str] = None
     samples: List[ReportQuerySample]
 
+    @staticmethod
+    def fix_string_list_values(
+        string_list: Optional[str], items_format: Union[str, int]
```
Comment (henrikstranneheim) [inline src/chanjo2/models/pydantic_models.py:238]:
```suggestion
        comma_sep_values: Optional[str], items_format: Union[str, int]
```

File: src/chanjo2/models/pydantic_models.py [hunk 3]
```diff
@@ -232,6 +233,42 @@ class ReportQuery(BaseModel):
...
+        if string_list is None:
+            return
+        if items_format == str:
+            return [item.strip() for item in string_list.split(",")]
+        else:
+            return [int(item.strip()) for item in string_list.split(",")]
+
+    @classmethod
+    def as_form(cls, form_data: FormData) -> "ReportQuery":
+        return cls(
```
Comment (henrikstranneheim) [inline src/chanjo2/models/pydantic_models.py:250]: Could this be assigned to some variables before the return, to make it more readable?

**Review by henrikstranneheim — APPROVED** [review]
Well done 👍

File: src/chanjo2/models/pydantic_models.py (hunk 3 above)
Comment (northwestwitch) [reply src/chanjo2/models/pydantic_models.py:250]: Sure, good idea!

File: src/chanjo2/models/pydantic_models.py [hunk 4]
```diff
@@ -232,6 +233,42 @@ class ReportQuery(BaseModel):
     case_display_name: Optional[str] = None
     samples: List[ReportQuerySample]
 
+    @staticmethod
+    def fix_string_list_values(
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply src/chanjo2/models/pydantic_models.py]: Renamed to `comma_sep_values_to_list`

File: src/chanjo2/models/pydantic_models.py (hunk 3 above)
Comment (northwestwitch) [reply src/chanjo2/models/pydantic_models.py]: Fixed

**Pull request merged**